### PR TITLE
Fix Javadoc build when in a release

### PR DESCRIPTION
**Pull request by awholegunch**
Porting of #8525 to 4.1.x

Running as a Docs Build and Test: https://builds.cask.co/browse/CDAP-DBT15-3

### CODE REVIEW

File: cdap-docs/build.sh [hunk 1]
```diff
@@ -77,8 +77,8 @@ function run_command() {
     docs-set )          build_docs_set; warnings=$?;;
     docs-set-local )    build_docs_set_local; warnings=$?;;
     docs-web-only )     build_docs_web_only; warnings=$?;;
-    javadocs )          build_javadocs docs;;
-    javadocs-all )      build_javadocs all;;
+    javadocs )          build_javadocs docs; warnings=$?;;
```
Comment (dereklwood) [inline cdap-docs/build.sh]: ``${?}`` for style-consistency (and lots more occurrences)

File: cdap-docs/build.sh [hunk 2]
```diff
@@ -198,36 +203,52 @@ function build_javadocs() {
...
+  fi
+  echo "========================================================"
+  echo "Building and installing CDAP to ${temp_repo}"
+  mvn clean install ${temp_repo} -P examples,templates,release -DskipTests -Dgpg.skip=true
+  errors=$?
+  if [[ ${errors} -eq 0 ]]; then
+    echo "========================================================"
+    echo "Building and installing CDAP finished; running Javadocs"
+    mvn ${javadoc_run} ${temp_repo} -DskipTests -Dgpg.skip=true -DisOffline=false ${debug_flag}
+    errors=$?
```
Comment (dereklwood) [inline cdap-docs/build.sh]: probably better to not shadow/reassign the same variable outside the conditional... ie, use a different variable here, and add a ``return`` in the error section below

File: cdap-docs/build.sh [hunk 3]
```diff
@@ -198,36 +203,52 @@ function build_javadocs() {
...
+    if [[ ${errors} -eq 0 ]]; then
+      echo
+      echo "Javadocs Build Start: ${start}"
+      echo "                 End: `date`"
+    else
+      echo "Error building Javadocs: ${errors}"
```
Comment (dereklwood) [inline cdap-docs/build.sh]: ``${errors}`` will be the numerical exit code, so this wont be a particularly helpful to print it here as if it were the actual errors

File: cdap-docs/build.sh [hunk 4]
```diff
@@ -198,36 +203,52 @@ function build_javadocs() {
...
+    fi
   else
-    echo "Error building Javadocs: ${warnings}"
+    echo "Error installing CDAP: ${errors}"
```
Comment (dereklwood) [inline cdap-docs/build.sh]: same as above ``Error installing CDAP: 1``

**Comment by dereklwood** [comment]
some minor comments, otherwise 👍 lgtm

File: cdap-docs/build.sh (hunk 1 above)
Comment (awholegunch) [reply cdap-docs/build.sh]: Fixed...and some $1s for good measure.

File: cdap-docs/build.sh (hunk 2 above)
Comment (awholegunch) [reply cdap-docs/build.sh]: fixed.

File: cdap-docs/build.sh (hunk 3 above)
Comment (awholegunch) [reply cdap-docs/build.sh]: fixed.

File: cdap-docs/build.sh (hunk 4 above)
Comment (awholegunch) [reply cdap-docs/build.sh]: fixed.

**Comment by awholegunch** [comment]
Re-running build:
Running as a Docs Build and Test: https://builds.cask.co/browse/CDAP-DBT15-3 (passes)

**Review by dereklwood — APPROVED** [review]
👍 lgtm

**Pull request merged**